File: Practica3/Pila.java
public class Pila {
    private float arreglo[];
    private int top;
    private int n;

    public Pila(int n) {
        this.top = 0;
        this.n = n;
        arreglo = new float[n + 1];
    }

    public void push(float e) {
        if (this.top == n) {
            System.out.println("Pila llena");
        } else {
            this.top = this.top + 1;
            arreglo[this.top] = e;
        }
    }

    public float pop() {
        if (this.top == 0) {
            System.out.println("Pila vacia");
            return 0;
        } else {
            float dato = arreglo[this.top];
            this.top = this.top - 1;
            return dato;
        }
    }

    public float peek() {
        return arreglo[this.top];
    }

    public boolean isEmpty() {
        if (this.top == 0) {
            return true;
        } else {
            return false;
        }
    }

    public boolean isFull() {
        if (this.top == n) {
            return true;
        } else {
            return false;
        }
    }
}

File: Practica3/Pila.py
class Pila:
    def __init__(self, n):
        self.top = 0
        self.n = n
        self.arreglo = [0] * (n + 1)

    def push(self, e):
        if self.top == self.n:
            print("Pila llena")
        else:
            self.top += 1
            self.arreglo[self.top] = e

    def pop(self):
        if self.top == 0:
            print("Pila vacia")
            return 0
        else:
            dato = self.arreglo[self.top]
            self.top -= 1
            return dato

    def peek(self):
        if self.top == 0:
            print("Pila vacia")
            return None
        return self.arreglo[self.top]

    def is_empty(self):
        return self.top == 0

    def is_full(self):
        return self.top == self.n